provider/openstack/provider.go

```diff
@@ -1017,20 +1017,40 @@ func (e *Environ) StartInstance(args environs.StartInstanceParams) (*environs.St
 		networks = append(networks, nova.ServerNetworks{NetworkId: networkId})
 	}
 
-	var apiPort int
-	if args.InstanceConfig.Controller != nil {
-		apiPort = args.InstanceConfig.Controller.Config.APIPort()
-	} else {
-		// All ports are the same so pick the first.
-		apiPort = args.InstanceConfig.APIInfo.Ports()[0]
-	}
-	groupNames, err := e.firewaller.SetUpGroups(args.ControllerUUID, args.InstanceConfig.MachineId, apiPort)
-	if err != nil {
-		return nil, errors.Annotate(err, "cannot set up groups")
+	// For BUG 1680787: openstack: add support for neutron networks where port security is disabled.
+	// If PortSecurityEnabled is false, don't create security groups, instance boot will fail.
+	createSecurityGroups := true
+	// there will be 1 or no networks in []networks based finding network code
```

Comment:
Technically yes, in practice, the openstack provider has no default networks defined and there is only expected to be one network listed as a config value for bootstrap.

Experimentation showed me that nova boot with 2 networks, one where port_security_enabled is true and one where it's false, cannot have a security group specified.  Interestingly the
instance details will show a security group of default.

I changed the code in case there is more than 1 network in the future.